tests/integration/context_api_test.py

```diff
@@ -0,0 +1,77 @@
+
+import os
+import tempfile
+from docker import errors
+from docker.context import ContextAPI
+from docker.tls import TLSConfig
+from .base import BaseAPIIntegrationTest
+
+
+class ContextLifecycleTest(BaseAPIIntegrationTest):
+    def test_lifecycle(self):
+        # check there is no context 'test'
+        ctx = ContextAPI.get_context()
+        assert ctx is not None and ctx.Name is "default"
+
+        ctx = ContextAPI.get_context("test")
+        assert ctx is None
+
+        ctx = ContextAPI.get_current_context()
+        assert ctx is not None
+
+        success = False
+        dirpath = tempfile.mkdtemp()
+        ca = tempfile.NamedTemporaryFile(
+            prefix=os.path.join(dirpath, "ca.pem"), mode="r")
+        cert = tempfile.NamedTemporaryFile(
+            prefix=os.path.join(dirpath, "cert.pem"), mode="r")
+        key = tempfile.NamedTemporaryFile(
+            prefix=os.path.join(dirpath, "key.pem"), mode="r")
+        # create context 'test
+        docker_tls = TLSConfig(
+            client_cert=(cert.name, key.name),
+            ca_cert=ca.name)
+        ContextAPI.create_context(
+            "test", tls_cfg=docker_tls)
+
+        # check for a context 'test' in the context store
+        contexts = ContextAPI.contexts()
+        for ctx in contexts:
+            if ctx.Name == "test":
+                success = True
+
+        assert success is True
```

Comment:
A mor pythonic way would be:
```suggestion
        assert any([ctx.Name == "test" for ctx in ContextAPI.contexts()])
```